src/fundus/parser/utility.py

```diff
@@ -453,7 +447,95 @@ def parse_urls_from_image(node: lxml.html.HtmlElement) -> Optional[Dict[str, str
         return None
 
 
-_relative_source_selector = XPath("./ancestor::picture/source")
+class _DimensionCalculator:
+    def __init__(
+        self, width: Optional[float] = None, height: Optional[float] = None, ratio: Optional[float] = None
+    ) -> None:
+        self.width = width
+        self.height = height
+        self.ratio = ratio
+
+    def calculate(
+        self, width: Optional[float] = None, height: Optional[float] = None, dpr: Optional[float] = None
+    ) -> Optional[Dimension]:
+        if not (width or height):
+            width = self.width
+            height = self.height
+        if dimension := Dimension.from_ratio(width, height, self.ratio):
+            return dimension * (dpr or 1)
+        return None
+
+
+_min_width_pattern = re.compile(r"min-width:\s*(?P<minwidth>[0-9]*)(?P<descriptor>[pxem]+)")
+_width_x_height_pattern = re.compile(r"(?P<width>[0-9]+)x(?P<height>[0-9]+)")
```

Comment:
It might make sense to also match `max-width`, which is sometimes used instead of `min-width` for smaller images.
e.g. https://www.zeit.de/gesundheit/2023-11/neues-alzheimer-medikament-zulassung-lecanemab-leqembi-demenz